### PR TITLE
feat: add query to get submissions

### LINKED ISSUE #510: Implement logic within the `ModSubmissionsStore` to fetch and distribute data
# Problem
We need to continue working on and implementing useful and readable code that can be used throughout the components in the Moderation Page. Continue working on that in the branch created in this draft request. It should show the correct number for each choice based on the data fetched. 

relates to #486 

# Success
Implements the store to fetch and filter the submissions
### Requirements

- [ ] Implement logic to fetch submission
- [ ] Implement filters for the fetched submissions

# Implementation notes
implement the filtering necessary for certain components in the store so it isn't handled in multiple places in many components. If you need to setup your local database follow the ReadMe from the [server](https://github.com/ourjapanlife/findadoc-server )

### PR DESCRIPTION
**Pull request by theyokohamalife**
Resolves #510 #537

## 🔧 What changed
1. The store now fetches data from the database.
2.  The data that was fetched is displayed in the Moderation Dashboard

## 🧪 Testing instructions

1.  Run your local backend server and db emulator.
2.  Run `yarn dev:localserver`
3. Go to the route `/moderation`

## 📸 Screenshots

-   ### Before
-
image

-   ### After

image

### CODE REVIEW

**Comment by NabbeunNabi** [comment]
This is currently blocked on prod due to unauthenticated users. But am making a note so that we can see it that we can still use the dev server to make sure it is working and make a PR

File: components/ModSubmissionListContainer.vue
Comment (ermish) [inline components/ModSubmissionListContainer.vue:21]: is there a reason this is a button vs a clickable div? 
From an accessibility perpective, this might be confusing as a button

File: components/ModSubmissionListContainer.vue
Comment (ermish) [inline components/ModSubmissionListContainer.vue:58]: ideal candidate for a switch statement

File: components/ModSubmissionListContainer.vue
Comment (ermish) [inline components/ModSubmissionListContainer.vue:70]: this is a locale string, not UTC, correct? 
maybe I would call this something like `convertDateToLocalTime`

File: stores/moderationSubmissionsStore.ts
Comment (ermish) [inline stores/moderationSubmissionsStore.ts:10]: nit: omg there's semicolons on almost EVERY line. 😿 😢 😭 🥹

File: stores/moderationSubmissionsStore.ts
Comment (ermish) [inline stores/moderationSubmissionsStore.ts:30]: rename to improve readability
```suggestion
            getSubmissionsGqlQuery,
```

**Review by ermish — CHANGES_REQUESTED** [review]
good design! few minor touch ups

File: components/ModSubmissionListContainer.vue
Comment (NabbeunNabi) [reply components/ModSubmissionListContainer.vue:21]: No reason. I really am still working on semantic HTML

File: stores/moderationSubmissionsStore.ts
Comment (NabbeunNabi) [reply stores/moderationSubmissionsStore.ts:10]: I will do my best to fix this by tomorrow

File: stores/moderationSubmissionsStore.ts
Comment (NabbeunNabi) [reply stores/moderationSubmissionsStore.ts:30]: I will do this. I will also refactor the gql to get all the data if that is ok?

File: components/ModSubmissionListContainer.vue
Comment (ermish) [inline components/ModSubmissionListContainer.vue:23]: I'm sure you're going to grumble at this one 😆, but since it's clickable we should add in the pointer curse option and maybe through in a nice hover effect ✨

**Review by ermish — CHANGES_REQUESTED** [review]
one more change! everything else looks great 🚀

**Review by ermish — APPROVED** [review]
Ship it!

**Review by Anissa3005 — CHANGES_REQUESTED** [review]
Before merging can you make your linting pass. So on your branch run the linting command and see what is failing, I think it's a typescript linting error since that one is working.

**Review by Anissa3005 — APPROVED** [review]
Linting is passing 🥳

**Review by ermish — APPROVED** [review]
Ship it! Again!

**Pull request merged**
